config/sql/config.go

```diff
@@ -93,4 +83,19 @@ func Configure(p *config.Provider) {
 		)
 		r.UseAsync = true
 	})
+	p.AddResourceConfigurator("azurerm_mssql_server_transparent_data_encryption", func(r *config.Resource) {
+		r.Version = common.VersionV1Alpha2
+		r.References = config.References{
+			"server_id": config.Reference{
+				Type:      "MSSQLServer",
+				Extractor: rconfig.ExtractResourceIDFuncPath,
+			},
+			"key_vault_key_id": config.Reference{
+				Type:      rconfig.APISPackagePath + "/keyvault/v1alpha2.Key",
```

Comment:
So far, we have preferred to have version constants so that with version bumps we can easily maintain corresponding references.